### ISSUE TITLE
[BUG] [RULES] Cannot match `System.out.println` with anything mores specifc than `pattern: "System*"`

### ISSUE THREAD

**Issue by JonahSussman** [konveyor/analyzer-lsp#693]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

N/A

### Priority

Undefined (Default)

### Current Behavior

There appears to be no combination of rules to match `System.out.println`, aside from:

```yaml
java.referenced:
    pattern: "System*"
```

I have tried setting `pattern` to each one of these, with and without the additional `location: METHOD_CALL` but they give no matches:

```yaml
- "java.io.PrintStream.println"
- "java.io.PrintStream.println(*)"
- "java.lang.System.out.println"
- "java.lang.System.out.println(*)"
- "*out*"
- "System.*"
- "System\\.*"
- "*System.out*"
- "System.out*"
```

### Expected Behavior

Any one of these more specific combinations should match the call of `System.out.println`.

### How Reproducible

Always (Default)

### Steps To Reproduce

On Fedora 40 with maven installed:

1. Navigate to an empty directory
1. [Download the latest release of Kantra ](https://github.com/konveyor/kantra?tab=readme-ov-file#linux)
2. Generate the helloworld project: `mvn archetype:generate -DgroupId=io.konveyor -DartifactId=helloworld -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.4 -DinteractiveMode=false`
3. Download the attached rule set.
4. Run `./kantra analyze --input helloworld --output analysis --rules blog-post-rules --skip-static-report --overwrite --enable-default-rulesets=false`
5. Note that note of the rules matched

[blog-post-rules.zip](https://github.com/user-attachments/files/16716178/blog-post-rules.zip)

### Environment

```markdown
- `be48b40abe11a6de1a9ce82dcd660eea430e8cf66a4caf91ebb54e2384eb5738  kantra`
```

### Anything else?

N/A
